Extension/src/Debugger/extension.ts

```diff
@@ -64,9 +75,91 @@ export async function initialize(context: vscode.ExtensionContext): Promise<void
+    disposables.push(vscode.commands.registerCommand('C_Cpp.removeSshTarget', removeSshTargetImpl));
+    disposables.push(vscode.commands.registerCommand(refreshCppSshTargetsView, (node?: BaseNode) => sshTargetsProvider.refresh(node)));
+    disposables.push(vscode.commands.registerCommand('C_Cpp.setActiveSshTarget', async (node: TargetLeafNode) => {
+        await setActiveSshTarget(node.name);
+        await vscode.commands.executeCommand(refreshCppSshTargetsView);
+    }));
+    disposables.push(vscode.commands.registerCommand('C_Cpp.selectSshTarget', selectSshTarget));
+    disposables.push(vscode.commands.registerCommand('C_Cpp.selectActiveSshTarget', async () => {
+        const name: string | undefined = await selectSshTarget();
+        if (name) {
+            await setActiveSshTarget(name);
+            await vscode.commands.executeCommand(refreshCppSshTargetsView);
+        }
+    }));
+    disposables.push(vscode.commands.registerCommand('C_Cpp.activeSshTarget', getActiveSshTarget));
+    disposables.push(sshTargetsProvider);
+    for (const sshConfig of getSshConfigurationFiles()) {
+        fileWatchers.push(chokidar.watch(sshConfig, {ignoreInitial: true})
+            .on('add', () => vscode.commands.executeCommand(refreshCppSshTargetsView))
+            .on('change', () => vscode.commands.executeCommand(refreshCppSshTargetsView))
+            .on('unlink', () => vscode.commands.executeCommand(refreshCppSshTargetsView)));
+    }
+    vscode.commands.executeCommand('setContext', 'showCppSshTargetsView', true);
 }
 
 export function dispose(): void {
+    // Don't wait
+    fileWatchers.forEach(watcher => watcher.close().then(() => {}, () => {}));
     disposables.forEach(d => d.dispose());
 }
+
+async function addSshTargetImpl(): Promise<string> {
+    const name: string | undefined = await vscode.window.showInputBox({
+        title: localize('enter.ssh.target.name', 'Enter SSH Target Name'),
+        placeHolder: localize('ssh.target.name.place.holder', 'Example: `mySSHTarget`'),
+        ignoreFocusOut: true
+    });
+    if (name === undefined) {
+        // Cancelled
+        return '';
+    }
+
+    const command: string | undefined = await vscode.window.showInputBox({
+        title: localize('enter.ssh.connection.command', 'Enter SSH Connection Command'),
+        placeHolder: localize('ssh.connection.command.place.holder', 'Example: `ssh hello@microsoft.com -A`'),
+        ignoreFocusOut: true
+    });
+    if (!command) {
+        return '';
+    }
+
+    const newEntry: { [key: string]: string } = sshCommandToConfig(command, name);
+
+    const targetFile: string | undefined = await vscode.window.showQuickPick(getSshConfigurationFiles().filter(file => pathAccessible(file, fs.constants.W_OK)), { title: localize('select.ssh.config.file', 'Select a SSH configuration file') });
+    if (!targetFile) {
+        return '';
+    }
+
+    const parsed: Configuration = await getSshConfiguration(targetFile, false);
```

Comment:
nit: rename `parsed` to the thing that is getting parsed. is it a config file?

Reply:
It's a json representing the entire ssh config file.